network/resolvconf.go

```diff
@@ -0,0 +1,78 @@
+package network
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"syscall"
+)
+
+const resolv = "/etc/resolv.conf"
+
+func InstallResolvconf(name, containerDNS string) error {
+	containerDNSResolv, err := touchXDGFile("dagger/net/" + name + "/resolv.conf")
+	if err != nil {
+		return err
+	}
+
+	// create the resolv.conf for the container namespace by swapping out the
+	// nameservers from the original, keeping any options and search domains
+	if err := replaceNameservers(containerDNS, containerDNSResolv); err != nil {
+		return fmt.Errorf("replace nameservers: %w", err)
+	}
+
+	upstreamResolv, err := touchXDGFile("dagger/net/" + name + "/resolv.conf.upstream")
```

Comment:
Just realized AppArmor isn't going to like this since now we'll have dynamic `resolv.conf` paths.

I guess I could move these to `/etc/dnsmasq.d/`? https://gitlab.com/apparmor/apparmor/-/blob/master/profiles/apparmor.d/usr.sbin.dnsmasq#L43

Reply:
Went with [paths](https://gitlab.com/apparmor/apparmor/-/blob/bba1a023bf146352f4d4d796fd0a8d1dd1e54fa7/profiles/apparmor.d/usr.sbin.dnsmasq#L110-116) specifically designated for the `dnsname` plugin that this PR vendors and cannibalizes. Hopefully they don't remove them, since `dnsname` is technically unmaintained now. All the more reason to have some sort of smoke test for this. :grimacing: